client/src/components/EventDetail.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { render, screen, waitFor } from '@testing-library/react';
+import EventDetail from './EventDetail';
+import type { PollingEvent } from './EventTypes';
+
+//  Mockujeme globální fetch
+global.fetch = vi.fn();
```

Comment:
tenhle radek bych smazal a udelal to trochu jinak, radek 25:

```
    beforeEach(() => {
      global.fetch = vi.fn();

    });

    afterEach(() => {
      vi.clearAllMocks();
    })
```

takto se ti ten mock nastavi pred kazdym testem a po kazdem testu se smaze. Je to cistsi reseni